### PR TITLE
Edit decks redux

### PR DESCRIPTION
**Pull request by alexmiddeleer**
Request for help trying to test this (and pen-test the RLS)

### CODE REVIEW

File: next/pages/deck-builder.js [hunk 1]
```diff
@@ -1,72 +1,54 @@
...
 import DeckBuilderSidebar from '../components/deck-builder-sidebar';
 import CardSearchFilters from '../components/card-search-filters';
 import SliderSwitch from '../components/slider-switch';
 import DeckBuilderCardDisplay from '../components/deck-builder-card-display';
+import ErrorMessage from '../components/error-message';
+import PropTypes from 'prop-types';
+import { useApolloClient } from 'react-apollo-hooks';
 
-const initialSearchFilters = {
+const init = {
```
Comment (lily-pytel) [inline next/pages/deck-builder.js]: Can we compromise on `initFilters`?

File: next/lib/deck-utils.js [hunk 2]
```diff
@@ -30,6 +35,160 @@ export const addCardToDeck = (deck, card) => {
...
+      });
+    })
+    .catch(reason => {
+      console.error(reason);
+      setIsError(true);
+    });
+};
+
+export const useStateDeck = deckId => {
+  const [deckInProgress, _setDeckInProgress] = useState(
```
Comment (lily-pytel) [inline next/lib/deck-utils.js]: I've never seen `useState` used outside a component. Not sure how I feel about it, but also the feelings are not very strong.

File: next/components/save-deck.js [hunk 3]
```diff
@@ -82,16 +72,13 @@ export default function SaveDeck({ deckInProgress, setDeckInProgress }) {
...
 
 SaveDeck.propTypes = {
+  deckId: PropTypes.number,
   deckInProgress: PropTypes.shape({
     deckName: PropTypes.string,
     deckPath: PropTypes.shape({
       id: PropTypes.number,
       name: PropTypes.string
     }),
-    deckPath: PropTypes.shape({
```
Comment (lily-pytel) [inline next/components/save-deck.js]: This is supposed to be `deckPower`. Sorry, I thought I fixed it in my previous PR.

File: next/pages/deck-builder.js (hunk 1 above)
Comment (alexmiddeleer) [reply next/pages/deck-builder.js]: Sounds good

File: next/lib/deck-utils.js (hunk 2 above)
Comment (alexmiddeleer) [reply next/lib/deck-utils.js]: Yeah, I know what you mean. I'll probably refactor this a bit more after I do some additional testing.

File: next/components/save-deck.js (hunk 3 above)
Comment (alexmiddeleer) [reply next/components/save-deck.js]: Lol I feel like I've also fixed this like 7 times and somehow it keeps coming back.

**Pull request merged**